src/sections/Meshery/Meshery-integrations/index.js

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { useState } from "react";
+import IntegrationsGrid from "./IntegrationsGrid";
+import IntegrationList from "../../../assets/data/integrations/IntegrationList";
+import Integrations from "./Integration.style";
+
+function index() {
+  const [categoryList, setcategoryList] = useState([
+    { name: "All", isSelected: true },
+    { name: "Platforms", isSelected: false },
+    { name: "Service Mesh", isSelected: false },
+    { name: "Operating System", isSelected: false },
+    { name: "Collaboration", isSelected: false },
+  ]);
+
+  const [integrationCollection, setintegrationCollection] =
+    useState(IntegrationList);
+
+  const filterCategory = (e) => {
+    const category = e.target.innerHTML;
+
+    if (category !== "All") {
+      categoryList[0].isSelected = false;
+      categoryList.map((item) =>
+        item.name === category ? (item.isSelected = !item.isSelected) : null
+      );
+      setcategoryList(categoryList);
+    } else {
+      categoryList[0].isSelected = true;
+
+      categoryList.map((item) =>
+        item.name !== "All" ? (item.isSelected = false) : null
+      );
+      setcategoryList(categoryList);
+    }
+
+    let tempCat = [];
+
+    categoryList.map((item) => {
+      if (item.isSelected) tempCat.push(item.name);
+    });
+
+    filterCollection(tempCat);
+  };
+
+  const filterCollection = (selectedCategoryList) => {
+    let tempInt = [];
+
+    IntegrationList.map((Integration) => {
+      selectedCategoryList.map((item) => {
+        if (item === Integration.category) {
+          tempInt.push(Integration);
+        } else if (item === "All") {
+          tempInt.push(Integration);
+        }
+      });
+    });
```

Comment:
I don't understand the logic here.
If all you are doing is pushing `Integration` to `tempInt` then why use `if..else`?

Reply:
Why use `map` when you are only looping through the list.
Use `forEach` instead

Reply:
I am using if..else because if someone selects any category, it will match it with the category in IntegrationList, and if it matches, then it will add it to the tempInt array.

But that's not the case for the "All" category cause we don't have any integration with a category all, so to return all the items, I am using an additional else statement.

Reply:
But in both the case you are pushing integration to the list, so why add a condition? And even if you need one then both can be merged into one single `if` condition using `&&` operator

Reply:
Now that you put it this way, I think I can merge the code using an `||` operator.